benchmarks/polys.py

```diff
@@ -341,4 +371,24 @@ class TimeGCD_QuadraticNonMonicGCD(_TimeGCD):
 
 class TimeGCD_SparseNonMonicQuadratic(_TimeGCD):
     GCDExampleCLS = _SparseNonMonicQuadratic
-    params = [(1, 3, 5), ('expr', 'dense', 'sparse')]
\ No newline at end of file
+    params = [(1, 3, 5), ('expr', 'dense', 'sparse')]
+
+
+class _TimeGaussianInt(_TimeOP):
+    """Benchmarks for GCDs method when input is Gaussian Integer"""
+
+    def expected(self, f, g, d, syms):
+        expected_gcd = gcd(f, g)
+
+        return expected_gcd
+
+    def get_func_expr(self, f, g, d, syms):
+        return lambda: gcd(f, g)
+
+    def get_func_poly(self, f, g, d):
+        return lambda: f.gcd(g)
+
+
+class TimeGCD_GaussInt(_TimeGaussianInt):
+    GCDExampleCLS = _GaussianInteger
+    params = [(1, 2, 3), ('expr', 'dense')]
```

Comment:
This should also time the sparse implementation because that is what is going to be changed.

Reply:
I added the sparse method but it is failing. Should I make the separate class and other functions for `Gaussian` inputs

Reply:
The code should be fixed so that it works. The `to_ring` method assumes that the domain is always `QQ` which will fail for Gaussian integers.

Reply:
What needs to be changed is the faulty assumption that the domain is `QQ` here:
https://github.com/sympy/sympy_benchmarks/blob/e4f235f2d890d99024f90e96a3ab83af9bb51760/benchmarks/polys.py#L33
The domain should not be hard-coded as `QQ`.